purchases_flutter_ui/android/src/main/kotlin/com/revenuecat/purchases_flutter_ui/PurchasesFlutterUiPlugin.kt

```diff
@@ -0,0 +1,92 @@
+package com.revenuecat.purchases_flutter_ui
+
+import android.app.Activity
+import android.util.Log
+import com.revenuecat.purchases.PurchasesErrorCode
+import com.revenuecat.purchases.hybridcommon.ui.PaywallResultListener
+import com.revenuecat.purchases.hybridcommon.ui.presentPaywallFromFragment
+import com.revenuecat.purchases.ui.revenuecatui.activity.PaywallResult
+import io.flutter.embedding.android.FlutterFragmentActivity
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.embedding.engine.plugins.activity.ActivityAware
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+
+class PurchasesFlutterUiPlugin: FlutterPlugin, MethodCallHandler, ActivityAware {
+    private val TAG = "PurchasesFlutter"
+
+    private var activity: Activity? = null
+
+    private lateinit var channel : MethodChannel
+
+    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
```

Comment:
In the existing plugin we also have the `registerWith` static method. That seems to be used for very old flutter versions, but we are not supporting those so I think it's ok to just have this.